inc/c_util/two_d_array_ll.h

```diff
@@ -15,10 +15,236 @@
+                    /* Codes_SRS_TWO_D_ARRAY_07_018: [ If there are any errors then TWO_D_ARRAY_ALLOCATE_NEW_ROW(T) shall fail and return a non-zero value. ]*/                              \
+                    LogError("malloc(sizeof(col)=%zu) failed", two_d_array->cols * sizeof(T));                                                                                               \
+                    result = MU_FAILURE;                                                                                                                                                     \
+                }                                                                                                                                                                            \
+                else                                                                                                                                                                         \
+                {                                                                                                                                                                            \
+                    (void)memcpy((void*)&two_d_array->row_arrays[row_index], &new_row, sizeof(T*));                                                                                          \
+                    result = 0;                                                                                                                                                              \
+                }                                                                                                                                                                            \
+            }                                                                                                                                                                                \
+            else                                                                                                                                                                             \
+            {                                                                                                                                                                                \
+                /* Codes_SRS_TWO_D_ARRAY_07_016: [ If the row specified by row_index has already been allocated, TWO_D_ARRAY_ALLOCATE_NEW_ROW(T) shall fail and return a non-zero value. ]*/ \
+                result = MU_FAILURE;                                                                                                                                                         \
+            }                                                                                                                                                                                \
```

Comment:
If this is really an error we should LogError here.
Also we usually put the error first, so consider reversing the logic in 189 to check for != NULL, log the error on the true branch and deal with the allocation on the false branch of the if.

Reply:
added the logError and changed the order